spec/rubocop/cop/internal_affairs/node_matcher_directive_spec.rb

```diff
@@ -222,5 +222,31 @@ class MyCop
         PATTERN
       RUBY
     end
+
+    it 'registers an offense when using class methods' do
+      expect_offense(<<~RUBY, method: method)
+        #{method} :"self.foo?", <<~PATTERN
+        ^{method}^^^^^^^^^^^^^^^^^^^^^^^^^ Precede `#{method}` with a `@!method` YARD directive.
+          (str)
+        PATTERN
+
+        #{method} "self.bar?", <<~PATTERN
+        ^{method}^^^^^^^^^^^^^^^^^^^^^^^^ Precede `#{method}` with a `@!method` YARD directive.
+          (str)
+        PATTERN
+      RUBY
+
+      expect_correction(<<~RUBY)
+        # @!method self.foo?(node)
```

Comment:
Isn't the expected behavior described in #12665 as follows?

```suggestion
        # @!method foo?(node)
        # @!scope class
```

It also seems necessary to implement the detection and autocorrection in the opposite direction:

```ruby
# @!method self.foo?(node)
# @!scope class
#{method} :"foo?", <<~PATTERN
  (str)
PATTERN
```

Reply:
You're right, I missed that part. I just looked at the PR in that repo. I changed it to correct to this syntax instead.

If there are variations I've missed to test against let me know.